core/trino-main/src/main/java/io/trino/execution/scheduler/FaultTolerantPartitioningSchemeFactory.java

```diff
@@ -51,43 +55,90 @@ public FaultTolerantPartitioningSchemeFactory(NodePartitioningManager nodePartit
 
     public FaultTolerantPartitioningScheme get(PartitioningHandle handle)
     {
-        return cache.computeIfAbsent(handle, this::create);
+        FaultTolerantPartitioningScheme result = cache.get(handle);
```

Comment:
why not `computeIfAbsent`?

Reply:
because of recursive call?

Reply:
Yeah, it throws a `ConcurrentModificationException` if a map is modified while in `computeIfAbsent`

Reply:
can you add a code comment. It looks unexpected and sooner or later someone will "fix" it :)